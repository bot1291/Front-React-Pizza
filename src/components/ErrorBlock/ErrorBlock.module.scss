@import '../../styles/variables.scss';

.error {
	display: flex;
	flex-direction: column;
	justify-content: center;
	align-items: center;
	gap: 5px;
	height: 60vh;
	text-align: center;
}

.text {
	font-size: 25px;
}

.image {
	width: 260px;
}
