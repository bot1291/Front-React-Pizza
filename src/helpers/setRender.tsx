import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { createReduxStore } from '../store/store';

// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
const setRender = <T,>(Component: JSX.Element, store?: Record<string, T>) => {
	return render(
		<Provider store={createReduxStore(store)}>{Component}</Provider>
	);
};

export default setRender;
