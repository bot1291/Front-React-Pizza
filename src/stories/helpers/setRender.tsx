import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import { createReduxStore } from '../../store/store';

const setRender = <T,>(
	Component: JSX.Element,
	store?: Record<string, T>
): JSX.Element => {
	return (
		<MemoryRouter>
			<Provider store={createReduxStore(store)}>{Component}</Provider>
		</MemoryRouter>
	);
};

export default setRender;
