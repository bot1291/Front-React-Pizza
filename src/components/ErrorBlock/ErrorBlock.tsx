import styles from './ErrorBlock.module.scss';
import { ErrorBlockProps } from './ErrorBlock.props';
import { FC } from 'react';
import cn from 'classnames';

export const ErrorBlock: FC<ErrorBlockProps> = ({ className, ...props }) => {
	return (
		<div className={cn(className, styles.error)} {...props}>
			<h1>500 Ошибка Сервера</h1>
			<img
				className={styles.image}
				src="https://i.imgur.com/F0qoSsb.png"
				alt="Pizza error"
			/>
			<span className={styles.text}>Не удалось загрузить пиццы.</span>
			<span className={styles.text}>
				Перезагрузите страничку или попробуйте позже
			</span>
		</div>
	);
};
