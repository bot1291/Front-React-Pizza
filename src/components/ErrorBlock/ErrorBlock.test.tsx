import setRender from '../../helpers/setRender';
import { screen } from '@testing-library/dom';
import { ErrorBlock } from './ErrorBlock';

describe('ErrorBlock', () => {
	it('Should follow the structure', () => {
		setRender(<ErrorBlock />);
		const wrapperElem = screen.getByTestId('error-wrapper');
		expect(wrapperElem).toMatchSnapshot();
	});
});
