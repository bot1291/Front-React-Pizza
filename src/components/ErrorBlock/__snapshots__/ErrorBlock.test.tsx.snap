// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorBlock Should follow the structure 1`] = `
<div
  class="error"
  data-testid="error-wrapper"
>
  <h1>
    500 Ошибка Сервера
  </h1>
  <img
    alt="Pizza error"
    class="image"
    src="https://i.imgur.com/F0qoSsb.png"
  />
  <span
    class="text"
  >
    Не удалось загрузить пиццы
  </span>
  <span
    class="text"
  >
    Перезагрузите страничку или попробуйте позже
  </span>
</div>
`;
